src/plugins/orangehrmRecruitmentPlugin/Api/InterviewAttachmentAPI.php

```diff
@@ -0,0 +1,282 @@
+        return new EndpointResourceResult(InterviewAttachmentModel::class, $interviewAttachment);
+    }
+
+    /**
+     * @param InterviewAttachment $interviewAttachment
+     */
+    private function setInterviewAttachment(InterviewAttachment $interviewAttachment)
+    {
+        $interviewAttachment->getDecorator()->setInterviewById(
+            $this->getRequestParams()->getInt(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_INTERVIEW_ID
+            )
+        );
+        $interviewAttachment->setComment(
+            $this->getRequestParams()->getStringOrNull(
+                RequestParams::PARAM_TYPE_BODY,
+                self::PARAMETER_COMMENT
+            )
+        );
+    }
+
+    /**
+     * @param InterviewAttachment $InterviewAttachment
+     */
+    private function setBase64Attachment(InterviewAttachment $InterviewAttachment): void
+    {
+        $base64Attachment = $this->getRequestParams()->getAttachmentOrNull(
+            RequestParams::PARAM_TYPE_BODY,
+            self::PARAMETER_ATTACHMENT
+        );
+        if (is_null($base64Attachment)) {
+            return;
+        }
+        $InterviewAttachment->setFileName($base64Attachment->getFilename());
+        $InterviewAttachment->setFileType($base64Attachment->getFileType());
+        $InterviewAttachment->setFileSize($base64Attachment->getSize());
+        $InterviewAttachment->setFileContent($base64Attachment->getContent());
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForCreate(): ParamRuleCollection
+    {
+        return new ParamRuleCollection(
+            new ParamRule(
+                self::PARAMETER_INTERVIEW_ID,
+                new Rule(Rules::ENTITY_ID_EXISTS, [Interview::class])
+            ),
+            new ParamRule(
+                self::PARAMETER_ATTACHMENT,
+                new Rule(
+                    Rules::BASE_64_ATTACHMENT,
+                    [
+                        null,
+                        null,
+                        self::PARAM_RULE_FILE_NAME_MAX_LENGTH
+                    ]
+                )
+            ),
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    self::PARAMETER_COMMENT,
+                    new Rule(Rules::STRING_TYPE),
+                    new Rule(Rules::LENGTH, [null, self::PARAM_RULE_COMMENT_MAX_LENGTH]),
+                ),
+                true
+            ),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function delete(): EndpointResult
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getValidationRuleForDelete(): ParamRuleCollection
+    {
+        throw $this->getNotImplementedException();
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $attachmentId = $this->getRequestParams()->getInt(
```

Comment:
seems can anyone access any attachment by id

Reply:
yes, Once user roles implemented, we can use ACCECCIBLE_ENTITY_IDS validation I guess

Reply:
yes, Once user roles implemented, we can use ACCECCIBLE_ENTITY_IDS validation I guess

Reply:
role based permisson added